docs/sources/operations/storage/retention.md

```diff
@@ -3,7 +3,157 @@ title: Retention
+
+Retention period is configure via the [`limits_config`](./../../../configuration/#limits_config) configuration section. It supports two type of retention `retention_stream` which is applied to all chunks matching the selector
+
+The example below can be added to your Loki configuration to configure global retention.
+
+```yaml
+...
+limits_config:
+  retention_period: 744h
+  retention_stream:
+  - selector: '{namespace="dev"}'
+    priority: 1
+    period: 24h
+  per_tenant_override_config: /etc/overrides.yaml
+...
+```
+
+Per tenant retention can be defined using the `/etc/overrides.yaml` files. For example:
+
+```yaml
+overrides:
+    "29":
+        retention_period: 168h
+        retention_stream:
+        - selector: '{namespace="prod"}'
+          priority: 2
+          period: 336h
+        - selector: '{container="loki"}'
+          priority: 1
+          period: 72h
+    "30":
+        retention_stream:
+        - selector: '{container="nginx"}'
+          priority: 1
+          period: 24h
+```
+
+The most specific matching rule is selected. The rule selection is as follow:
+
+- If a per-tenant `retention_stream` matches the current stream, the highest priority is picked.
+- Otherwise if a global `retention_stream` matches the current stream, the highest priority is picked.
+- Otherwise if a per-tenant `retention_period` is specified it will be applied.
+- The the global `retention_period` will be selected if nothing else matched.
+- And finally if no global `retention_period` is set, the default `30days (744h)` retention is selected.
+
+Stream matching is using the same syntax as Prometheus label matching:
+
+- `=`: Select labels that are exactly equal to the provided string.
+- `!=`: Select labels that are not equal to the provided string.
+- `=~`: Select labels that regex-match the provided string.
+- `!~`: Select labels that do not regex-match the provided string.
+
+The two configurations will, for example, set those rules:
+
+- All tenants except `29` and `30` will have the retention for streams that are in the `dev` namespace to `24h` hours, and other streams will have a retention of `744h`.
+  - For tenant `29`:
+    - All streams except, the container `loki` or the `namespace` prod, will have retention of 1 week (`168h`).
+    - All streams in `prod` will have a retention of `336h` (2 weeks), even the if the container label is also `loki` since the priority of the `prod` rule is higher.
+    - Steams that have the label container `loki` but are not in the namespace `prod` will have 72h retention.
+  - For tenant `30`:
+    - All streams, except those having the label container `nginx` will have a retention of `744h` (the global retention,since it doesn't overrides a new one).
+    - Streams that have the label `nginx` will have a retention of `24h`.
+
+> The minimum retention period is 24h
```

Comment:
Revise sentence formatting and add a period at the end of the sentence.

The minimum retention period is `24h`.

(I think that this sentence belongs earlier in the document. A good spot might be just after line 91.)